### PR TITLE
Fixed #33048 -- Doc'd that DEBUG static files requests don't use middleware chain.

### PR DESCRIPTION
**Pull request by valleyofblackpanther**
ticket-33048

Refactored Django's documentation to explain the role of middleware
chain and how the exclusion of middleware helps the process of serving
the static files automatically.

### CODE REVIEW

File: docs/docs/conf.py/conf.py.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Configuration file for the Sphinx documentation builder.
+#
+# Full list of options can be found in the Sphinx documentation:
+# https://www.sphinx-doc.org/en/master/usage/configuration.html
+
+#
+# -- Project information -----------------------------------------------------
+#
+
+project = "sphinx-autobuild"
```
Comment (carltongibson) [inline docs/docs/conf.py/conf.py.txt:10]: These changes are unrelated. Please drop them.

File: docs/howto/static-files/index.txt [hunk 2]
```diff
@@ -108,6 +108,16 @@ do this by adding the following snippet to your urls.py::
...
+.. note:: 
+     
+    During development, we know that serving files could be done automatically;
+    it doesn't go through the middleware chain when we set the DEBUG is True. 
+    What it does is that it enables a view function that serves our static
+    files automatically. But it's not the whole process. Runserver usually
+    uses WSGI_APPLICATION, which in turn depends on middleware functions. 
+    But when Debug is true, runserver overrides the core functionality per
+    se. And it adds an automatic serving of static files; the file serving 
+    doesn't go through MIDDLEWARE.
```
Comment (carltongibson) [inline docs/howto/static-files/index.txt:120]: I don't think going into the exact implementation is called for… 🤔

I had something in mind more like... 

```
    Static files are served via a wrapper at the WSGI application layer, and
    static files requests do not pass through the normal middleware chain. 
```

I think this could added to the existing `note` block immediately above, perhaps beginning, `Finally, …` 🤔 

(Also the block should have a single blank line following it for formatting.)

**Review by carltongibson — COMMENTED** [review]
Hi @valleyofblackpanther — thanks for this. I've left a few suggestions.

File: docs/howto/static-files/index.txt (hunk 2 above)
Comment (valleyofblackpanther) [reply docs/howto/static-files/index.txt:120]: I'll make sure to add the following lines. Mr Carlton, I know you are busy at the moment because of the Django 4.0 release. But when I was working with this ticket. I saw a similar ticket #27325 [https://code.djangoproject.com/ticket/27325](url). I'll summarize this for you in two lines. **Offer a solution for static file serving suitable for production.** But people tried this five years ago and they nearly came with a solution but didn't really apply it. You know what is best for Django's growth, do you think this will add any value to Django.

File: docs/docs/conf.py/conf.py.txt (hunk 1 above)
Comment (valleyofblackpanther) [reply docs/docs/conf.py/conf.py.txt:10]: I'll make sure to delete the following lines. Thank you.

File: docs/howto/static-files/index.txt (hunk 2 above)
Comment (carltongibson) [reply docs/howto/static-files/index.txt]: Hi @valleyofblackpanther If ticket-27325 were to be resolved, we'd remove the added text here as part of that. 

But, as you note, it's been a few years already. It may be a few more yet, so this note is intended to help folks in the meantime. (I'd guess that'd be @kezabelle's take opening the issue…)

Comment (valleyofblackpanther) [reply docs/howto/static-files/index.txt]: Hi @carltongibson. So you are meaning to tell that @kezabelle is responsible to open the issue again.

**Review by hannseman — COMMENTED** [review]
`docs/docs/conf.py/conf.py.txt` is still present in the commit. Running `git rm  docs/docs/conf.py/conf.py.txt` should do the trick.

File: docs/howto/static-files/index.txt [hunk 3]
```diff
@@ -108,6 +108,9 @@ do this by adding the following snippet to your urls.py::
     folder; it doesn't perform static files discovery like
     :mod:`django.contrib.staticfiles`.
 
+    Finally, Static files are served via a wrapper at the WSGI application layer, and
```
Comment (hannseman) [inline docs/howto/static-files/index.txt:111]:
```suggestion
    Finally, static files are served via a wrapper at the WSGI application layer, and
```

File: docs/howto/static-files/index.txt (hunk 2 above)
Comment (kezabelle) [reply docs/howto/static-files/index.txt]: I'm pleased to see someone picking up this documentation ticket, so thanks :)

Carlton's correct: for my purposes (i.e: [ticket 33048](https://code.djangoproject.com/ticket/33048) being worked on here) documenting the special-casing is enough (actually for my purposes I just monkeypatch'd some stuff, and now the problem no longer exists for me ;)).

The other ticket, [27325](https://code.djangoproject.com/ticket/27325) is not my call, but would one day be good to finally see land (personally if it were _me_, a starting point would be to convert `StaticFilesHandlerMixin` into a middleware, which isn't insurmountably difficult, and promote static-files serving to be a MW, and then later co-ordinate introducing WhiteNoise to replace _that_, YMMV).

Comment (valleyofblackpanther) [reply docs/howto/static-files/index.txt]: Okay, thanks for the instruction @kezabelle. Will look into that.

**Comment by felixxm** [comment]
@valleyofblackpanther Thanks :+1: Welcome aboard :boat: 

@kezabelle Do you want to take another look?

**Comment by kezabelle** [comment]
@felixxm yep, LGTM. I _could_ stand to see the `middleware chain` linked to either the `MIDDLEWARE` setting or the middleware docs, but I don't think it matters overly; the additional admonition is informative enough that (had I been looking at the docs ;)) I'd have noted it when I needed to work around it.

**Comment by felixxm** [comment]
> @felixxm yep, LGTM. I _could_ stand to see the `middleware chain` linked to either the `MIDDLEWARE` setting or the middleware docs, but I don't think it matters overly; the additional admonition is informative enough that (had I been looking at the docs ;)) I'd have noted it when I needed to work around it.

Thanks :+1: I added link to the middleware topic.
